### ISSUE TITLE
Unable to read rows enclosed in double quotes

### ISSUE THREAD

**Issue by odiaseo** [thephpleague/csv#420]
## Bug Report

_(Fill in the relevant information below to help triage your issue.)_

| Information | Description |
|--------------|---------|
| Version | 9.7.0 |
| PHP version | 7.4 |
| OS Platform |  MacOs, Ubuntu|

### Summary
_(Please explain in plain english your bug)_
Given  a csv file with the data below, the reader does not display the records as expected
"this,",is
the,data£

### Standalone code, or other way to reproduce the problem

_(Please complete the text below to help us fix the issue)_

### Expected result

_(What was the expected (correct) behavior?)_
['this,', 'is']
### Actual result
['"this,, ',is\n']
_(What is the current (buggy) behavior?)_

## Checks before submitting

* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/thephpleague/csv/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/thephpleague/csv/pulls)
* [ ] I have added every step to reproduce the bug.
* [ ] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by nyamsprod** [comment]
@odiaseo could you provide a reproducible code as am I unable to do it myself.

```php
<?php

$str = <<<CSV
"this,",is
the,data£
CSV;

$csv = Reader::createFromString($str);
foreach ($csv as $record) {
    var_export($record);
    echo PHP_EOL;
}
```

returns the following

```bash
array (
  0 => 'this,',
  1 => 'is',
)
array (
  0 => 'the',
  1 => 'data£',
)

``` 

Which is the expected behaviour

**Comment by odiaseo** [comment]
Creating the from path using the files:
[RA-8212_bom.csv](https://github.com/thephpleague/csv/files/6286945/RA-8212_bom.csv)
[RA-8212_bom_cr.csv](https://github.com/thephpleague/csv/files/6286946/RA-8212_bom_cr.csv)

**Comment by nyamsprod** [comment]
@odiaseo I still don't have the reproducible script

**Comment by odiaseo** [comment]
```       
         use League\Csv\Reader;

        $reader = Reader::createFromPath($filename);
        $record = $reader->fetchOne(0);

        var_dump($record);
```
 Is the above ok, where the $filename is the file above

**Comment by nyamsprod** [comment]
@odiaseo I believe the issue is not with the package but with your file encoding charset did you already check it ?

**Comment by odiaseo** [comment]
Is the library not able to handle different charsets?

**Comment by odiaseo** [comment]
I had to overwrite the League\Csv\Stream class in order to replace the rewind function to fix this bug. See below.
```    /**
     * Rewind the file to the first line.
     * @throws UnavailableFeature
     */
    public function rewind(): void
    {
        if (!$this->is_seekable) {
            throw UnavailableFeature::dueToMissingStreamSeekability();
        }

        rewind($this->stream);
        fseek($this->stream, 0);

        $firstBytes = fread($this->stream, 3);
        if ("\xEF\xBB\xBF" !== $firstBytes) {
            fseek($this->stream, 0);
        }
        $this->offset = 0;
        $this->value = false;
        if (0 !== ($this->flags & SplFileObject::READ_AHEAD)) {
            $this->current();
        }
    }```

**Comment by nyamsprod** [comment]
@odiaseo in the documentation it is stated that you should either use https://csv.thephpleague.com/9.0/connections/filters/

or 

https://csv.thephpleague.com/9.0/converter/charset/

Depending on your preferences and/or context

**Comment by odiaseo** [comment]
It would be helpful if you can give a working code example of how to read those files attached as opposed to using the workaround to overwrite those Stream class. Thanks.

**Comment by nyamsprod** [comment]
@odiaseo sorry this is a issue tracker since it is not an issue you can otherwise ask for help in dedicated forum.
